### DISCUSSION TITLE
Get wrong color in getColors function

### DISCUSSION THREAD

**Discussion by gisjiashu** [iTwin/itwinjs-core#1080] [Q&A] [answered]
Hi ,

I'd like to consult the TTRRGGBB format of color convert in ColorDef.ts. When I set color number to 0x000000FF, it displayed RED color instead of BLUE color. It can produced by color number 0x00FF0000.

In the line 224([https://github.com/imodeljs/imodeljs/blob/master/core/common/src/ColorDef.ts]), the blue and red return value should be switched.

![Wrong Color Dislplay](https://user-images.githubusercontent.com/6380896/113076167-1206f300-9201-11eb-8aa9-16e718441798.png)

**Comment by kabentley** [comment, marked as answer]
Please see [this discussion](https://github.com/imodeljs/imodeljs/blob/b477bd91dfe3a77cd72ed97c8676f06d164955bb/core/common/src/ColorDef.ts#L26-L42) about the format ColorDef uses. ColorDef stores as 0xTTBBGGRR, which is different than the color format used in HTML. Color 0xff is red in ColorDef format.

**Reply by pmconne** [reply]
@gisjiashu you can [create a ColorDef from an HTML or CSS color string](https://github.com/imodeljs/imodeljs/blob/b477bd91dfe3a77cd72ed97c8676f06d164955bb/core/common/src/ColorDef.ts#L59); e.g., the following produces red: `ColorDef.create("#ff0000")`.

**Reply by gisjiashu** [reply]
hi 
I reviewed the ColorDef.create comment, it's correct defined in this function. I just confused the TTRRGGBB and TTBBGGRR.
It's greate that we can use string or number to defined color.

Thanks for your patient answer.

**Reply by kabentley** [reply]
yes, all of this is very confusing and unfortunate (the swapping of red and blue vs. the HTML/css format). Unfortunately it cannot be changed since these values are stored in iModels. All we can do is document it as clearly as possible.
